### ISSUE TITLE
rook cluster helm chart no longer deploys an ingress on k8s 1.17

### ISSUE THREAD

**Issue by TomHellier** [rook/rook#9174]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
It should be possible to deploy to k8s 1.17 with an ingress.

**Expected behavior:**
Deploying the helm chart using argocd 2.1 fails to deploy an ingress resource due to an error introduced in rook v1.7.3 with https://github.com/rook/rook/pull/8666

> .Capabilities.APIVersions.Has "networking.k8s.io/v1"

evaluating to true on 1.17 due to networking.k8s.io/v1 containing the [NetworkPolicy](https://v1-17.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#networkpolicy-v1-networking-k8s-io)

see similar issue elsewhere on the internet:
https://gitea.com/gitea/helm-chart/issues/77

I think it would be good to provide similar options elsewhere like other opensource charts in the helm ecosystem, so you can override this setting entirely.

**Environment**:
* OS (e.g. from /etc/os-release): n/a
* Kernel (e.g. `uname -a`): n/a
* Cloud provider or hardware configuration: n/a
* Rook version (use `rook version` inside of a Rook Pod): 1.7.7
* Storage backend version (e.g. for ceph do `ceph -v`): n/a
* Kubernetes version (use `kubectl version`): 1.17
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): baremetal
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest/ceph-toolbox.html)): n/a

**Comment by travisn** [comment]
@TomHellier What is your values.yaml? The Rook CI runs on various K8s versions back to 1.15 and the helm tests are currently passing. For example, on the latest release-1.7 build, the helm test is passing on 1.16 as seen [here](https://github.com/rook/rook/actions/runs/1468112794). 

Perhaps it is related to the helm version? The Rook CI is using v3.6.2. What version are you using?

**Comment by TomHellier** [comment]
@travisn from my recollection of the ceph helm tests, they deploy the cephcluster crds etc, but do not deploy an ingress.

I think this is because the minikube in the integration tests does not support this, I spent some time yesterday investigating what would be required to get some integration test coverage for this bug, I think we can add an ingress addon to minikube when it starts, and define an ingress [as shown in this commit on my fork]( https://github.com/TomHellier/rook/commit/63d5a843336af6fe219c3c33e99929c52c3d2247)

I ran out of time yesterday and was not able to look at it today. I'll do some further investigations tomorrow.
